components/feature/session/src/main/java/mozilla/components/feature/session/SwipeRefreshFeature.kt

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.session
+
+import android.view.View
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.EngineView
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+
+/**
+ * Feature implementation to add pull to refresh functionality to browsers.
+ *
+ * @param swipeRefreshLayout Reference to SwipeRefreshLayout that has an [EngineView] as its child.
+ */
+class SwipeRefreshFeature(
+    private val sessionManager: SessionManager,
+    private val swipeRefreshLayout: SwipeRefreshLayout
+) : SelectionAwareSessionObserver(sessionManager), LifecycleAwareFeature,
+    SwipeRefreshLayout.OnChildScrollUpCallback, SwipeRefreshLayout.OnRefreshListener {
+
+    init {
+        swipeRefreshLayout.setOnRefreshListener(this)
+        swipeRefreshLayout.setOnChildScrollUpCallback(this)
+    }
+
+    /**
+     * Start feature: Starts adding pull to refresh behavior for the active session.
+     */
+    override fun start() {
+        observeSelected()
```

Comment:
This should support both modes: Selected session or a fixed session id (compare with other features). Background is that in activities that display a custom tab (or a progressive web app later on) we have a specific fixed session id that we display instead of following the selected session (like in the default browser activity).

Feel free to file a follow-up issue. Not something you'd have to do in this PR.

Reply:
I'll add an optional `sessionId` parameter